fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/CoordinatorService.java

```diff
@@ -161,13 +164,39 @@ public CompletableFuture<CreateTableResponse> createTable(CreateTableRequest req
             sanityCheckPartitionedTable(tableDescriptor);
         }
 
+        MergeEngine mergeEngine = tableDescriptor.getMergeEngine();
+        if (mergeEngine != null) {
+            checkMergeEngine(mergeEngine, tableDescriptor.getSchema());
+        }
+
         // then create table;
         metadataManager.createTable(
                 tablePath, tableDescriptor, tableAssignment, request.isIgnoreIfExists());
 
         return CompletableFuture.completedFuture(response);
     }
 
+    private void checkMergeEngine(MergeEngine mergeEngine, Schema schema) {
+        if (mergeEngine.getType() == MergeEngine.Type.VERSION) {
+            String column = mergeEngine.getColumn();
+            RowType rowType = schema.toRowType();
+            int fieldIndex = rowType.getFieldIndex(column);
+            if (fieldIndex == -1) {
+                throw new IllegalArgumentException(
```

Comment:
throw `InvalidTableException` just like what we do while validating partitioned table..
And add a test to verify invalidate version column will throw exception..

Reply:
fixed